### PR TITLE
Feature/skale 1642 delegation functionality

### PR DESCRIPTION
**Pull request by dmitry-tk**
Add methods and contracts required for basic delegation functionality

### CODE REVIEW

**Comment by kladkogex** [comment]
Build fails

File: skale/contracts/delegation/validator_service.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    def ls(self):
+        """Returns list of registered validators.
+
+        :returns: List of validators
+        :rtype: list
+        """
+        validators = []
+        number_of_validators = self.number_of_validators()
+        for val_id in range(1, number_of_validators+1):
```
Comment (badrogger) [inline skale/contracts/delegation/validator_service.py]: I think we should use enumerate here. And may be list comprehension would be better in this case

File: skale/contracts/constants_holder.py [hunk 2]
```diff
@@ -20,14 +20,15 @@
...
 
 
-class Constants(BaseContract):
+class ConstantsHolder(BaseContract):
     @transaction_method
     def set_periods(self, new_reward_period, new_delta_period):
         op = self.contract.functions.setPeriods(new_reward_period, new_delta_period)
-        tx = post_transaction(self.skale.wallet, op, GAS['set_periods'])
-        return {'tx': tx}
+        tx_hash = post_transaction(self.skale.wallet, op, GAS['set_periods'])
```
Comment (badrogger) [inline skale/contracts/constants_holder.py]: Not sure, but may be it's better to make post_transaction return TxRes instance

Comment (dmitry-tk) [reply skale/contracts/constants_holder.py]: make sense

File: skale/contracts/delegation/validator_service.py (hunk 1 above)
Comment (dmitry-tk) [reply skale/contracts/delegation/validator_service.py]: done

**Pull request merged**